crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -67,69 +67,40 @@ pub enum State {
     Error,
 }
 
-struct SyncServiceInner {
-    room_list_service: Arc<RoomListService>,
-    encryption_sync_service: Arc<EncryptionSyncService>,
-    state: SharedObservable<State>,
-    encryption_sync_permit: Arc<AsyncMutex<EncryptionSyncPermit>>,
-    /// Supervisor task ensuring proper termination.
-    ///
-    /// This task is waiting for a `TerminationReport` from any of the other two
-    /// tasks, or from a user request via [`Self::stop()`]. It makes sure
-    /// that the two services are properly shut up and just interrupted.
-    ///
-    /// This is set at the same time as the other two tasks.
-    supervisor_task: Option<JoinHandle<()>>,
+/// A supervisor that starts two sync tasks, one for the room list and one for
+/// the end-to-end encryption support.
+struct SyncTaskSupervisor {
+    /// The task that supervises the two sync tasks.
+    task: JoinHandle<()>,
     /// `TerminationReport` sender for the [`Self::stop()`] function.
     ///
     /// This is set at the same time as all the tasks in [`Self::start()`].
-    supervisor_sender: Option<Sender<TerminationReport>>,
-}
-
-pub struct SyncService {
-    inner: Arc<AsyncMutex<SyncServiceInner>>,
-
-    /// Room list service used to synchronize the rooms state.
-    room_list_service: Arc<RoomListService>,
-
-    /// What's the state of this sync service? This field is replicated from the
-    /// [`SyncServiceInner`] struct, but it should not be modified in this
-    /// struct. It's re-exposed here so we can subscribe to the state
-    /// without taking the lock on the `inner` field.
-    state: SharedObservable<State>,
-
-    /// Global lock to allow using at most one [`EncryptionSyncService`] at all
-    /// times.
-    ///
-    /// This ensures that there's only one ever existing in the application's
-    /// lifetime (under the assumption that there is at most one [`SyncService`]
-    /// per application).
-    encryption_sync_permit: Arc<AsyncMutex<EncryptionSyncPermit>>,
+    abortion_sender: Sender<TerminationReport>,
```

Comment:
Can this be called `termination_sender`, with respect to `TerminatinoReport`? (that could be a `TerminationRequest` too)